pkg/cloud/ibmcloud/actuators/machine/actuator.go

```diff
@@ -231,24 +231,20 @@ func (ic *IBMCloudClient) Update(ctx context.Context, cluster *clusterv1.Cluster
 	}
 
 	if guest.Domain == nil ||
-		guest.MaxMemory == nil ||
 		guest.OperatingSystemReferenceCode == nil ||
-		guest.LocalDiskFlag == nil ||
+		guest.SupplementalCreateObjectOptions.FlavorKeyName == nil ||
```

Comment:
please also check `guest.SupplementalCreateObjectOptions == nil` before checking `guest.SupplementalCreateObjectOptions.FlavorKeyName`. We do not sure IBM Cloud behavior. It makes code solid to checking all pontetial erros in the condition.